marshal.go

```diff
@@ -444,18 +426,18 @@ func (dm *DependencyMap) extract(m map[string]interface{}) error {
 	dm.Names = make(map[string][]string)
 	dm.Schemas = make(map[string]*Schema)
 	for k, p := range m {
-		switch p.(type) {
+		switch p := p.(type) {
```

Comment:
Does this witch need a default to error out if it hits a type it doesn't expect?  Right now it's just ignoring them, but I can't tell if that's intentional or not.